### PR TITLE
Add changelog

### PR DESCRIPTION
**Pull request by rossf7**
Fixes https://github.com/thegreenwebfoundation/grid-intensity-go/issues/58

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+- Add heading with version and release date. e.g. `## 0.1.0 2022-06-21`
+- Push a git tag with the version you wish to release.
 
 ```
-git tag -a v0.1.0 -m "Initial release"
+git tag -a v0.1.0 -m "v0.1.0"
 ```
 
+- Once the release is created edit the description to add the changelog text.
+- Note: The release description will already have the commits added by GoReleaser.
```
Comment (rossf7) [inline docs/release.md:12]: There are some manual steps here but I don't think they are too onerous and we don't create that many releases.

See https://github.com/thegreenwebfoundation/grid-intensity-go/releases/tag/v0.3.0 for the current release description created by GoReleaser with the commits.

**Comment by mrchrisadams** [comment]
hey Ross, there's a setting here that is new to me.

I've been rebasing by default, but this makes me think you'd prefer we squash commits in future instead.

Screenshot 2022-08-24 at 11 46 04

Can you let me know which you prefer and the basis for it?

I don't have strong opinions on this really, but it would be good to document the basis whatever decision we take.

**Comment by rossf7** [comment]
@mrchrisadams PTAL when you have time.

I've re-enabled rebase merging. I think there is some benefit to squashing commits but I'm also fine with rebasing. 

If you think only squashing is worthwhile I can do a docs PR for it as a separate change.

**Comment by mrchrisadams** [comment]
Oops - sorry about the slow response on this one @rossf7.

I've developed a habit of trying to tidy up commit history for PRs with `git rebase --interactive`, but to be honest, I'm not 100% consistent using it.

Given the choice, I think I'd lean towards interactive rebasing, as it's been the default so far, but I'm happy to revisit later as a deliberate decision.

Either way, Im' happy with this and I'm merging the code in, as I think it's a separate issue to adding the changelog  :)

**Pull request merged**